internal/manifest/parse_test.go

```diff
@@ -0,0 +1,80 @@
+package manifest
+
+import (
+	"archive/tar"
+	"bytes"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestPathExtractor_untarLayer(t *testing.T) {
+	tempDir, err := os.MkdirTemp("", "untar-test")
```

Comment:
Please think about extracting the following code into function, like `createTar` or something similar.